spec/services/hubee_api_client_spec.rb

```diff
@@ -0,0 +1,203 @@
+RSpec.describe HubEEAPIClient do
+  let(:hubee_api_client) { described_class.new }
+
+  before do
+    stub_request(:post, hubee_api_client.send(:hubee_auth_url))
+      .to_return(status: 200, headers: { 'Content-Type' => 'application/json' }, body: { "access_token" => "access_token" }.to_json)
+  end
+
+  let(:access_token) { 'access_token' }
+
+  describe '#get_organization' do
+    subject(:get_organization) { hubee_api_client.get_organization(siret, code_commune) }
+
+    let(:hubee_organization_payload) { {
+      "country" => "France",
+      "code" => nil,
+      "postalCode" => "92140",
+      "type" => "SI",
+      "companyRegister" => "21920023500014",
+      "branchCode" => "92023",
+      "phoneNumber" => "0123456789",
+      "name" => "COMMUNE DE CLAMART",
+      "email" => "example@admin.fr",
+      "territory" => "CLAMART",
+      "status" => "ACTIF"
+    } }
+
+    context 'when organization already exist' do
+      let(:siret) { '21920023500014' }
+      let(:code_commune) { '75001' }
+
+      before do
+        stub_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
+          .with(
+            headers: { 'Authorization' => "Bearer #{access_token}" })
+          .to_return(status: 200, body: hubee_organization_payload.to_json, headers: { 'Content-Type' => 'application/json' })
+      end
+
+      it 'calls the stubbed request' do
+        get_organization
+
+        expect(a_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
+                 .with(headers: { 'Authorization' => "Bearer #{access_token}" })
+        ).to have_been_requested
+      end
+
+      it 'returns the hubee organization payload' do
+        expect(get_organization).to eq(hubee_organization_payload)
+      end
+    end
+
+    context 'when organization does not exist' do
+      let(:siret) { '21050136700010' }
+      let(:code_commune) { '05245' }
+
+      context 'and returns a 404 not found error' do
+        before do
+          stub_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
+            .with(
+              headers: { 'Authorization' => "Bearer #{access_token}" })
+            .to_return(status: 404, body: '', headers: { 'Content-Type' => 'application/json' })
+        end
+
+        it 'raises a Faraday::ResourceNotFound error' do
+          expect { get_organization }.to raise_error(Faraday::ResourceNotFound)
+        end
+      end
```

Comment:
C'est le même contexte, 404 veut dire "not found"

```suggestion
    context 'when organization does not exist' do
      let(:siret) { '21050136700010' }
      let(:code_commune) { '05245' }

      before do
        stub_request(:get, "#{hubee_api_client.send(:api_host)}/referential/v1/organizations/SI-#{siret}-#{code_commune}")
          .with(
            headers: { 'Authorization' => "Bearer #{access_token}" })
          .to_return(status: 404, body: '', headers: { 'Content-Type' => 'application/json' })
      end

      it 'raises a Faraday::ResourceNotFound error' do
        expect { get_organization }.to raise_error(Faraday::ResourceNotFound)
      end
    end
```